.github/workflows/on_pull_request.yml

```diff
@@ -0,0 +1,60 @@
+name: Build & Deploy
```

Comment:
I would suggest: `Build & Review`, `Build & Staging`  and `Build & Production` or.. `Build & Deploy Review`